src/packages/core/efuns_main.cc

```diff
@@ -351,12 +351,22 @@ void f_capitalize(void) {
 #endif
 
 #ifdef F_CHILDREN
+//TODO:fix this. I basically moved code from the old children into this function but it should probably
+//be some sort of helper in array.h which works for all types. In future this "fix" probably won't be 
+//if I make changes to array_t.
 void f_children(void) {
-  array_t *vec;
+  auto max_array_size { CONFIG_INT(__MAX_ARRAY_SIZE__ ) };
+  auto v { ObjectTable::get()->children(sp->u.string) };
+  auto len { v.size() < max_array_size ? v.size() : max_array_size };
+  auto res {allocate_empty_array(len)};
 
-  vec = children(sp->u.string);
+  for(auto i {0};i < v.size() && i < len;++i) {
```

Comment:
why do you keep insisting on using {} initializer?  it's not with style of other code ,just use int i = 0 here, it is much more clear